diff --git a/coreapi/v24.0/roothash/api/api.go b/coreapi/v24.0/roothash/api/api.go
index 180852d..8c9e4c0 100644
--- a/coreapi/v24.0/roothash/api/api.go
+++ b/coreapi/v24.0/roothash/api/api.go
@@ -199,7 +199,7 @@ type ExecutorCommittedEvent struct {
 // ExecutionDiscrepancyDetectedEvent is an execute discrepancy detected event.
 type ExecutionDiscrepancyDetectedEvent struct {
 	// Round is the round in which the discrepancy was detected.
-	Round uint64 `json:"round"`
+	Round *uint64 `json:"round"`
 	// Rank is the rank of the transaction scheduler.
 	Rank uint64 `json:"rank"`
 	// Timeout signals whether the discrepancy was due to a timeout.
