diff --git a/coreapi/v21.1.1/registry/api/api.go b/coreapi/v21.1.1/registry/api/api.go
index a0b48650..6f38ceaf 100644
--- a/coreapi/v21.1.1/registry/api/api.go
+++ b/coreapi/v21.1.1/registry/api/api.go
@@ -6,9 +6,9 @@ import (
 	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
 	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
 	"github.com/oasisprotocol/oasis-core/go/common/entity"
-	"github.com/oasisprotocol/oasis-core/go/common/node"
 
 	"github.com/oasisprotocol/nexus/coreapi/v21.1.1/consensus/api/transaction"
+	"github.com/oasisprotocol/nexus/coreapi/v22.2.11/common/node" // Borrow the 22.2.11 version of `node` because the 21.1.1 version is identical for serde purposes
 )
 
 // ModuleName is a unique module name for the registry module.
diff --git a/coreapi/v21.1.1/registry/api/runtime.go b/coreapi/v21.1.1/registry/api/runtime.go
index 8169c215..7bbc385c 100644
--- a/coreapi/v21.1.1/registry/api/runtime.go
+++ b/coreapi/v21.1.1/registry/api/runtime.go
@@ -8,13 +8,13 @@ import (
 	"github.com/oasisprotocol/oasis-core/go/common/cbor"
 	"github.com/oasisprotocol/oasis-core/go/common/crypto/hash"
 	"github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
-	"github.com/oasisprotocol/oasis-core/go/common/node"
 	"github.com/oasisprotocol/oasis-core/go/common/quantity"
 	"github.com/oasisprotocol/oasis-core/go/common/version"
 	storage "github.com/oasisprotocol/oasis-core/go/storage/api"
 
 	scheduler "github.com/oasisprotocol/nexus/coreapi/v21.1.1/scheduler/api"
 	staking "github.com/oasisprotocol/nexus/coreapi/v21.1.1/staking/api"
+	"github.com/oasisprotocol/nexus/coreapi/v22.2.11/common/node" // Borrow the 22.2.11 version of `node` because the 21.1.1 version is identical for serde purposes
 )
 
 // removed var block
