diff --git a/coreapi/v22.2.11/roothash/api/message/message.go b/coreapi/v22.2.11/roothash/api/message/message.go
index 702a27a6..06ff264f 100644
--- a/coreapi/v22.2.11/roothash/api/message/message.go
+++ b/coreapi/v22.2.11/roothash/api/message/message.go
@@ -4,6 +4,7 @@ package message
 import (
        "github.com/oasisprotocol/oasis-core/go/common/cbor"

+       governance "github.com/oasisprotocol/nexus/coreapi/v22.2.11/governance/api"
        registry "github.com/oasisprotocol/nexus/coreapi/v22.2.11/registry/api"
        staking "github.com/oasisprotocol/nexus/coreapi/v22.2.11/staking/api"
 )
@@ -12,6 +13,9 @@ import (
 type Message struct {
        Staking  *StakingMessage  `json:"staking,omitempty"`
        Registry *RegistryMessage `json:"registry,omitempty"`
+       // XXX: Added by Nexus. Backported from Eden so that this structure can
+       // represent all oasis-core versions.
+       Governance *GovernanceMessage `json:"governance,omitempty"`
 }

 // ValidateBasic performs basic validation of the runtime message.
@@ -42,3 +46,14 @@ type RegistryMessage struct {

 // ValidateBasic performs basic validation of the runtime message.
 // removed func
+
+// GovernanceMessage is a governance message that allows a runtime to perform governance operations.
+type GovernanceMessage struct {
+       cbor.Versioned
+
+       CastVote       *governance.ProposalVote    `json:"cast_vote,omitempty"`
+       SubmitProposal *governance.ProposalContent `json:"submit_proposal,omitempty"`
+}
+
+// ValidateBasic performs basic validation of a governance message.
+// removed func
