diff --git a/coreapi/v23.0/common/node/node.go b/coreapi/v23.0/common/node/node.go
index faabace1..4e274610 100644
--- a/coreapi/v23.0/common/node/node.go
+++ b/coreapi/v23.0/common/node/node.go
@@ -105,7 +105,24 @@ type nodeV2 struct { // nolint: maligned
 }
 
 // ToV3 returns the V3 representation of the V2 node descriptor.
-// removed func
+func (nv2 *nodeV2) ToV3() *Node {
+	nv3 := &Node{
+		Versioned:       cbor.NewVersioned(3),
+		ID:              nv2.ID,
+		EntityID:        nv2.EntityID,
+		Expiration:      nv2.Expiration,
+		P2P:             nv2.P2P,
+		Consensus:       nv2.Consensus,
+		Runtimes:        nv2.Runtimes,
+		SoftwareVersion: nv2.SoftwareVersion,
+		Roles:           nv2.Roles & ^roleReserved3,      // Clear consensus-rpc role.
+		TLS:             TLSInfo{PubKey: nv2.TLS.PubKey}, // Migrate to new TLS Info.
+	}
+	if nv2.VRF != nil {
+		nv3.VRF = *nv2.VRF
+	}
+	return nv3
+}
 
 // SoftwareVersion is the node's oasis-node software version.
 type SoftwareVersion string
