diff --git a/coreapi/v23.0/consensus/api/submission.go b/coreapi/v23.0/consensus/api/submission.go
index 78e7d8d2..a762e967 100644
--- a/coreapi/v23.0/consensus/api/submission.go
+++ b/coreapi/v23.0/consensus/api/submission.go
@@ -1,11 +1,8 @@
 package api
 
 import (
-	"sync"
 	"time"
 
-	staking "github.com/oasisprotocol/nexus/coreapi/v23.0/staking/api"
-	"github.com/oasisprotocol/oasis-core/go/common/logging"
 	"github.com/oasisprotocol/oasis-core/go/common/quantity"
 )
 
@@ -33,17 +30,7 @@ type noOpPriceDiscovery struct{}
 
 // SubmissionManager is a transaction submission manager interface.
 // removed interface
-
-type submissionManager struct {
-	backend        ClientBackend
-	priceDiscovery PriceDiscovery
-	maxFee         quantity.Quantity
-
-	noncesLock sync.Mutex
-	nonces     map[staking.Address]uint64
-
-	logger *logging.Logger
-}
+// removed type
 
 // Implements SubmissionManager.
 // removed func
