diff --git a/coreapi/v23.0/consensus/api/api.go b/coreapi/v23.0/consensus/api/api.go
index 43e73727..f8ae18bb 100644
--- a/coreapi/v23.0/consensus/api/api.go
+++ b/coreapi/v23.0/consensus/api/api.go
@@ -161,7 +161,12 @@ type Vote struct {
 // StatusState is the concise status state of the consensus backend.
 type StatusState uint8
 
-// removed var block
+var (
+	// StatusStateReady is the ready status state.
+	StatusStateReady StatusState
+	// StatusStateSyncing is the syncing status state.
+	StatusStateSyncing StatusState = 1
+)
 
 // String returns a string representation of a status state.
 func (s StatusState) String() string {
